azure-mgmt-resources/src/main/java/com/azure/management/AzureServiceClient.java

```diff
@@ -85,10 +92,52 @@ public String userAgent() {
 
     private SerializerAdapter serializerAdapter = new AzureJacksonAdapter();
 
+    private String sdkName;
+
+    /**
+     * Gets serializer adapter for JSON serialization/de-serialization.
+     *
+     * @return the serializer adapter.
+     */
     public SerializerAdapter getSerializerAdapter() {
         return this.serializerAdapter;
     }
 
+    /**
+     * Gets default client context.
+     *
+     * @return the default client context.
+     */
+    public Context getContext() {
+        Context context = new Context("java.version", JAVA_VERSION);
+        if (!CoreUtils.isNullOrEmpty(OS_NAME)) {
+            context = context.addData("os.name", OS_NAME);
+        }
+        if (!CoreUtils.isNullOrEmpty(OS_VERSION)) {
+            context = context.addData("os.version", OS_VERSION);
+        }
+        if (sdkName == null) {
+            String packageName = this.getClass().getPackage().getName();
+            if (packageName.endsWith(".models")) {
+                sdkName = packageName.substring(0, packageName.length() - ".models".length());
+            }
+        }
+        context = context.addData("Sdk-Name", sdkName);
+        context = context.addData("Sdk-Version", SDK_VERSION);

```

Comment:
I remember there's limitation on lenght of userAgent at different service, such as cosmos. how long is length of our fixed useragent string?

Reply:
A typical one is such `User-Agent:azsdk-java-com.azure.management.resources/2.0.0 (13.0.1; Windows 10 10.0)`

Since user can set ApplicatinId, there is no limit on the length. Though we can truncate that ApplicationId is needed. @ChenTanyi 
https://github.com/Azure/azure-libraries-for-java/blob/vnext/azure-mgmt-resources/src/main/java/com/azure/management/UserAgentPolicy.java#L112-L117

track1 is not much to compare since Azure SDK gives new guideline for it.
https://azure.github.io/azure-sdk/general_azurecore.html#telemetry-policy

For reference track1 `Azure-SDK-For-Java/null OS:Windows 10/10.0 MacAddressHash:9e21c3cc3783e949641d36ffd6a2e0c027267de76b62dcc951a57438fbc15f36 Java:13.0.1 (ResourceManagementClient, 2019-08-01)`

Reply:
thanks Weidong for detail info. According to azure core tememetry, shouldn't sdk package name and version  set by azure.core?

Reply:
@Tanyi "DO enforce that the application ID is no more than 24 characters in length. " from azure core guideline, again, shouldn't this be done in azure.core/azure.core.mgmt?

Reply:
@yungezz The azure core has a User-Agent policy, but it could only send constant UA, we have raised an issue and pr about it. But the azure-core team advise us to implement out own. Meanwhile, I think currently all code in `com.azure.management` could moved into azure-core-mgmt.

Reply:
Personally I do do not see how azure.core could get sdk package name and sdk version without we setting it to azure.core in the first place. Maybe I misunderstood the question.

Reply:
so `package-name`, `pacakge-version` isn't in the constant UA? then that makes a lot of sense. Thanks both to help clarify!

Reply:
These are in the UA, but it cannot be changed between request. That 
 is not we need. We use a restclient for every package. So we need to change UA during request. https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core/src/main/java/com/azure/core/http/policy/UserAgentPolicy.java#L84